File: object/src/main/kotlin/com/jessyt/object/chapter4/sub2/Page.java
package com.jessyt.object.chapter4.sub2;

import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.IOException;

/**
 * Created by LYT to 2021/10/05
 */
public class Page {
    public byte[] content(File file) throws IOException {
        byte[] array = new byte[1000];
        new FileInputStream(file).read(array);
        return array;
    }

    public int length(File file) throws Exception {
        try {
            return content(file).length;
        } catch (IOException e) {
            throw new Exception("계산이 불가능하다.", e);
        }
    }

    public int length(File file) throws IOException {
        return content(file).length;
    }

    public String content() throws IOException {
        int attempt = 0;
        while (true) {
            try {
                return http();
            } catch (IOException ex) {
                if (attempt >= 2) {
                    throw ex;
                }
            }
        }
    }

    @RetryOnFailure(attempts = 3)
    public String content() throws IOException {
        return http();
    }
}

File: object/src/main/kotlin/com/jessyt/object/chapter4/sub2/Page.kt
package com.jessyt.`object`.chapter4.sub2

import java.io.File
import java.io.FileInputStream

/**
 * Created by LYT to 2021/10/05
 */
class Page {
    fun length1(file: File): Int {
        if (!file.exists())
            throw IllegalArgumentException("file doesn't exist")
        return content1(file).length
    }

    fun content1(file: File): String {
        return "file content"
    }

    fun use() {
        val file = File("/path/file")
        length1(file)
    }

    fun length(file: File): Int {
        return content(file).size
    }

    fun content(file: File): ByteArray {
        return FileInputStream(file).readAllBytes()
    }
}

